executable/elf.ksy

```diff
@@ -486,6 +487,23 @@ types:
             type: u8
           - id: size
             type: u8
+      note_section:
+        seq:
+          - id: entries
+            type: note_section_entry
+            repeat: eos
+      note_section_entry:
+        seq:
+          - id: name_size
+            type: u4
+          - id: description_size
+            type: u4
+          - id: note_type
+            type: u4
+          - id: note_name
+            size: "name_size % 4 == 0 ? name_size: name_size + (4 - name_size % 4)"
+          - id: note_description
+            size: "description_size % 4 == 0 ? description_size: description_size + (4 - description_size % 4)"
```

Comment:
1. Put spaces around both `?` and `:` operators of the ternary operation.
2. If it's not justified to use double quotes `"` (e.g. to be able to use character escapes), use single `'` quotes instead to enclose KS expressions. A potential problem with double quotes `"` is that any [escape sequences](https://yaml.org/spec/1.2/spec.html#id2776092) used in the expression will be interpreted by the YAML parser, not the Kaitai Struct compiler. YAML recognizes some sequences that KSC doesn't, so that can create inconsistency between using `value: '"foo"'` and `value: "'foo'"` syntax.
   For example `value: "'\U00000041'"` will work (it is interpreted by the YAML parser and the compiler only gets the decoded string), but `value: '"\U00000041"'` will not (interpreted by KSC).
3. BTW, you can simplify it to:
   
   ```suggestion
             - id: note_name
               size: name_size + (-name_size % 4)
             - id: note_description
               size: description_size + (-description_size % 4)
   ```
   
   We make extra effort to ensure that modulo of negative integers works just the way the [**modulo** operation](https://en.wikipedia.org/wiki/Modulo_operation) is intended to (see the test [`expr_mod.ksy`](https://github.com/kaitai-io/kaitai_struct_tests/blob/97d68e9847aea0783eee21c27933ebf01fd4e441/formats/expr_mod.ksy#L1), which passes in all languages we're testing except Nim - see https://ci.kaitai.io/ and search test `ExprMod`), so you can safely use it.

   See discussion in https://github.com/kaitai-io/kaitai_struct_formats/pull/409#discussion_r566748681 on this alignment calculation approach.